Assignment on Classification Do the following in the iris dataset

Read the dataset to python environment
Do the necessary pre-processing steps
Find out which classification model gives the best result (try all the classification algorithms discussed in the session)

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.preprocessing import MinMaxScaler

In [2]:
from google.colab import files
files.upload()

Saving iris.csv to iris.csv


{'iris.csv': b'SL,SW,PL,PW,Classification\r\n5.1,3.5,1.4,0.2,Iris-setosa\r\n4.9,3,1.4,0.2,Iris-setosa\r\n,3.2,1.3,0.2,Iris-setosa\r\n4.6,3.1,1.5,0.2,Iris-setosa\r\n5,3.6,1.4,0.2,Iris-setosa\r\n5.4,3.9,1.7,0.4,Iris-setosa\r\n4.6,3.4,1.4,0.3,Iris-setosa\r\n5,3.4,1.5,0.2,Iris-setosa\r\n4.4,2.9,1.4,0.2,Iris-setosa\r\n4.9,3.1,1.5,0.1,Iris-setosa\r\n5.4,,1.5,0.2,Iris-setosa\r\n4.8,3.4,1.6,0.2,Iris-setosa\r\n4.8,3,1.4,0.1,Iris-setosa\r\n4.3,3,1.1,0.1,Iris-setosa\r\n5.8,4,1.2,0.2,Iris-setosa\r\n5.7,4.4,1.5,0.4,Iris-setosa\r\n5.4,3.9,1.3,0.4,Iris-setosa\r\n5.1,3.5,1.4,0.3,Iris-setosa\r\n5.7,3.8,1.7,0.3,Iris-setosa\r\n5.1,3.8,1.5,0.3,Iris-setosa\r\n5.4,3.4,1.7,0.2,Iris-setosa\r\n,3.7,1.5,0.4,Iris-setosa\r\n4.6,3.6,1,0.2,Iris-setosa\r\n5.1,3.3,1.7,0.5,Iris-setosa\r\n4.8,3.4,1.9,0.2,Iris-setosa\r\n5,3,1.6,0.2,Iris-setosa\r\n5,3.4,1.6,0.4,Iris-setosa\r\n5.2,,1.5,0.2,Iris-setosa\r\n5.2,3.4,,0.2,Iris-setosa\r\n4.7,3.2,1.6,0.2,Iris-setosa\r\n4.8,3.1,1.6,0.2,Iris-setosa\r\n5.4,3.4,1.5,0.4,Iris-setosa\r

In [3]:
df = pd.read_csv('iris.csv')

In [4]:
df.head()

SL   SW   PL   PW Classification
0  5.1  3.5  1.4  0.2    Iris-setosa
1  4.9  3.0  1.4  0.2    Iris-setosa
2  NaN  3.2  1.3  0.2    Iris-setosa
3  4.6  3.1  1.5  0.2    Iris-setosa
4  5.0  3.6  1.4  0.2    Iris-setosa

In [5]:
df.shape

(150, 5)

In [6]:
df.columns

Index(['SL', 'SW', 'PL', 'PW', 'Classification'], dtype='object')

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   SL              143 non-null    float64
 1   SW              144 non-null    float64
 2   PL              144 non-null    float64
 3   PW              150 non-null    float64
 4   Classification  150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


There are 150 total rows. But SL, SW and PL shows 143,144 and 144 respectively. Which means missing of data.

In [8]:
#basic details of the dataset
df.describe()

SL          SW          PL          PW
count  143.000000  144.000000  144.000000  150.000000
mean     5.855944    3.049306    3.756250    1.198667
std      0.828168    0.430644    1.761306    0.763161
min      4.300000    2.000000    1.000000    0.100000
25%      5.100000    2.800000    1.600000    0.300000
50%      5.800000    3.000000    4.350000    1.300000
75%      6.400000    3.300000    5.100000    1.800000
max      7.900000    4.400000    6.900000    2.500000

In [9]:
#Null value checking 
df.isna().sum()

SL                7
SW                6
PL                6
PW                0
Classification    0
dtype: int64

In [10]:
# Drop rows with missing value
df = df.dropna()

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 131 entries, 0 to 149
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   SL              131 non-null    float64
 1   SW              131 non-null    float64
 2   PL              131 non-null    float64
 3   PW              131 non-null    float64
 4   Classification  131 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.1+ KB


the data set is reduced to 131 rows from 150 rows. that is missing of data. So create dummy variables.

In [13]:
#Creating dummy variables
dummies = []
cols = ['SL', 'SW', 'PL']
for col in cols:
   dummies.append(pd.get_dummies(df[col]))

In [14]:
iris_dummies = pd.concat(dummies, axis=1)

In [15]:
df = df.drop(['SL', 'SW', 'PL'], axis=1)

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 131 entries, 0 to 149
Data columns (total 2 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   PW              131 non-null    float64
 1   Classification  131 non-null    object 
dtypes: float64(1), object(1)
memory usage: 3.1+ KB


In [17]:
X = df.values
y = df['Classification'].values

In [18]:
X = np.delete(X, 1, axis=1)

Divide the Data Set Into Training Data and Test Data

In [21]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=0)

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 131 entries, 0 to 149
Data columns (total 2 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   PW              131 non-null    float64
 1   Classification  131 non-null    object 
dtypes: float64(1), object(1)
memory usage: 3.1+ KB
